src/mcelog.c

```diff
@@ -210,95 +213,105 @@ static int socket_reinit(socket_adapter_t *self) {
-                                              mr.corrected_err_total) < 0) {
-    ERROR("%s: add corrected errors meta data failed", MCELOG_PLUGIN);
+                                              mr->corrected_err_total) < 0) {
+    ERROR(MCELOG_PLUGIN ": add corrected errors meta data failed");
     plugin_notification_meta_free(n->meta);
     return (-1);
   }
   if (plugin_notification_meta_add_signed_int(
-          n, "corrected memory timed errors", mr.corrected_err_timed) < 0) {
-    ERROR("%s: add corrected timed errors meta data failed", MCELOG_PLUGIN);
-    plugin_notification_meta_free(n->meta);
-    return (-1);
-  }
-  if (plugin_notification_meta_add_string(n, "corrected errors time period",
-                                          mr.corrected_err_timed_period) < 0) {
-    ERROR("%s: add corrected errors period meta data failed", MCELOG_PLUGIN);
+          n, "corrected memory timed errors", mr->corrected_err_timed) < 0) {
+    ERROR(MCELOG_PLUGIN ": add corrected timed errors meta data failed");
     plugin_notification_meta_free(n->meta);
     return (-1);
   }
+  if (mr->corrected_err_timed_period[0] != '\0')
+    if (plugin_notification_meta_add_string(n, "corrected errors time period",
+                                            mr->corrected_err_timed_period) <
+        0) {
+      ERROR(MCELOG_PLUGIN ": add corrected errors period meta data failed");
+      plugin_notification_meta_free(n->meta);
+      return (-1);
+    }
   if (plugin_notification_meta_add_signed_int(n, MCELOG_UNCORRECTED_ERR,
-                                              mr.uncorrected_err_total) < 0) {
-    ERROR("%s: add corrected errors meta data failed", MCELOG_PLUGIN);
-    plugin_notification_meta_free(n->meta);
-    return (-1);
-  }
-  if (plugin_notification_meta_add_signed_int(
-          n, "uncorrected memory timed errors", mr.uncorrected_err_timed) < 0) {
-    ERROR("%s: add corrected timed errors meta data failed", MCELOG_PLUGIN);
+                                              mr->uncorrected_err_total) < 0) {
+    ERROR(MCELOG_PLUGIN ": add corrected errors meta data failed");
     plugin_notification_meta_free(n->meta);
     return (-1);
   }
-  if (plugin_notification_meta_add_string(n, "uncorrected errors time period",
-                                          mr.uncorrected_err_timed_period) <
-      0) {
-    ERROR("%s: add corrected errors period meta data failed", MCELOG_PLUGIN);
+  if (plugin_notification_meta_add_signed_int(n,
+                                              "uncorrected memory timed errors",
+                                              mr->uncorrected_err_timed) < 0) {
+    ERROR(MCELOG_PLUGIN ": add corrected timed errors meta data failed");
     plugin_notification_meta_free(n->meta);
     return (-1);
   }
+  if (mr->uncorrected_err_timed_period[0] != '\0')
+    if (plugin_notification_meta_add_string(n, "uncorrected errors time period",
```

Comment:
If you reverse the condition and return early, you can get rid of the nested indentation. Alternatively, combine the two conditions and && them into a single if.